[ENHANCEMENT] react-formio@1.x should support React v16

## Environment

- **react-formio branch/tag:**
  - [x] 1.x
- NodeJS version: v12.7.0
- npm version: 6.10.0

## Steps to Reproduce

1. Create/open a React project which uses `react@16.*`
2. Install `formio@1x` as a production dependency on that project.
3. Compile/transpile and execute that React project inside a browser.

## Expected behavior

The project should be able to execute successfully inside a browser without any errors.

## Observed behavior

The project fails to execute. Following error shows on the browser console:

```console
./node_modules/react-formio/lib/components/FormComponents/address.js
Uncaught TypeError: _react2.default.createClass is not a function
```

@randallknutson I have limited the `test` script to work on `src` and `test` folders only. Ran tests before and after my PR #235. There are failures in both cases - both failures different from each other.

Here are the results. Before my PR:

```console
D:\GitHub\formio\react-formio>npm test

> react-formio@1.5.25 test D:\GitHub\formio\react-formio
> mocha --timeout 30000 --reporter spec test/.setup.js src/**/*.spec.js test/**/*.spec.js

internal/modules/cjs/loader.js:628
    throw err;
    ^

Error: Cannot find module 'react-widgets/lib/DropdownList'
Require stack:
- D:\GitHub\formio\react-formio\src\components\FormComponents\mixins\selectMixin.jsx
- D:\GitHub\formio\react-formio\src\components\FormComponents\address.jsx
- D:\GitHub\formio\react-formio\src\components\FormComponents\address.spec.js
- D:\GitHub\formio\react-formio\node_modules\mocha\lib\mocha.js
- D:\GitHub\formio\react-formio\node_modules\mocha\index.js
- D:\GitHub\formio\react-formio\node_modules\mocha\bin\_mocha
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:625:15)
    at Function.Module._load (internal/modules/cjs/loader.js:527:27)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (D:/GitHub/formio/react-formio/src/components/FormComponents/mixins/selectMixin.jsx:2:1)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at loader (D:\GitHub\formio\react-formio\node_modules\babel-register\lib\node.js:144:5)
    at Object.require.extensions.<computed> [as .jsx] (D:\GitHub\formio\react-formio\node_modules\babel-register\lib\node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (D:/GitHub/formio/react-formio/src/components/FormComponents/address.jsx:3:1)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at loader (D:\GitHub\formio\react-formio\node_modules\babel-register\lib\node.js:144:5)
    at Object.require.extensions.<computed> [as .jsx] (D:\GitHub\formio\react-formio\node_modules\babel-register\lib\node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (D:/GitHub/formio/react-formio/src/components/FormComponents/address.spec.js:4:1)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at loader (D:\GitHub\formio\react-formio\node_modules\babel-register\lib\node.js:144:5)
    at Object.require.extensions.<computed> [as .js] (D:\GitHub\formio\react-formio\node_modules\babel-register\lib\node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at D:\GitHub\formio\react-formio\node_modules\mocha\lib\mocha.js:250:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (D:\GitHub\formio\react-formio\node_modules\mocha\lib\mocha.js:247:14)
    at Mocha.run (D:\GitHub\formio\react-formio\node_modules\mocha\lib\mocha.js:576:10)
    at Object.<anonymous> (D:\GitHub\formio\react-formio\node_modules\mocha\bin\_mocha:637:18)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:788:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:840:10)
    at internal/main/run_main_module.js:17:11 {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\GitHub\\formio\\react-formio\\src\\components\\FormComponents\\mixins\\selectMixin.jsx',
    'D:\\GitHub\\formio\\react-formio\\src\\components\\FormComponents\\address.jsx',
    'D:\\GitHub\\formio\\react-formio\\src\\components\\FormComponents\\address.spec.js',
    'D:\\GitHub\\formio\\react-formio\\node_modules\\mocha\\lib\\mocha.js',
    'D:\\GitHub\\formio\\react-formio\\node_modules\\mocha\\index.js',
    'D:\\GitHub\\formio\\react-formio\\node_modules\\mocha\\bin\\_mocha'
  ]
}
npm ERR! Test failed.  See above for more details.
```

After my PR:

```console
D:\GitHub\akaustav\react-formio>npm test

> react-formio@1.5.25 test D:\GitHub\akaustav\react-formio
> mocha --timeout 30000 --reporter spec test/.setup.js src/**/*.spec.js test/**/*.spec.js

Warning: ReactTestUtils has been moved to react-dom/test-utils. Update references to remove this warning.
D:\GitHub\akaustav\react-formio\node_modules\formiojs\lib\formio.js:1368
Formio.Headers = Headers;
                 ^

ReferenceError: Headers is not defined
    at Object.<anonymous> (D:\GitHub\akaustav\react-formio\node_modules\formiojs\lib\formio.js:1368:18)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at Module._extensions..js (internal/modules/cjs/loader.js:788:10)
    at Object.require.extensions.<computed> [as .js] (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:152:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (D:\GitHub\akaustav\react-formio\node_modules\formiojs\lib\formio.form.js:23:15)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at Module._extensions..js (internal/modules/cjs/loader.js:788:10)
    at Object.require.extensions.<computed> [as .js] (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:152:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (D:\GitHub\akaustav\react-formio\node_modules\formiojs\lib\index.js:8:15)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at Module._extensions..js (internal/modules/cjs/loader.js:788:10)
    at Object.require.extensions.<computed> [as .js] (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:152:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (D:/GitHub/akaustav/react-formio/src/components/Formio.jsx:3:1)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at loader (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:144:5)
    at Object.require.extensions.<computed> [as .jsx] (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (D:\GitHub\akaustav\react-formio\src\components\index.js:7:15)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at loader (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:144:5)
    at Object.require.extensions.<computed> [as .js] (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (D:/GitHub/akaustav/react-formio/src/components/FormComponents/columns.jsx:2:1)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at loader (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:144:5)
    at Object.require.extensions.<computed> [as .jsx] (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at Object.<anonymous> (D:/GitHub/akaustav/react-formio/src/components/FormComponents/columns.spec.js:4:1)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at loader (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:144:5)
    at Object.require.extensions.<computed> [as .js] (D:\GitHub\akaustav\react-formio\node_modules\babel-register\lib\node.js:154:7)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at D:\GitHub\akaustav\react-formio\node_modules\mocha\lib\mocha.js:220:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (D:\GitHub\akaustav\react-formio\node_modules\mocha\lib\mocha.js:217:14)
    at Mocha.run (D:\GitHub\akaustav\react-formio\node_modules\mocha\lib\mocha.js:469:10)
    at Object.<anonymous> (D:\GitHub\akaustav\react-formio\node_modules\mocha\bin\_mocha:404:18)
    at Module._compile (internal/modules/cjs/loader.js:777:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:788:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Function.Module.runMain (internal/modules/cjs/loader.js:840:10)
    at internal/main/run_main_module.js:17:11
npm ERR! Test failed.  See above for more details.
```

--- a/lib/components/FormioGrid.js
+++ b/lib/components/FormioGrid.js
@@ -6,6 +6,10 @@ Object.defineProperty(exports, "__esModule", {
 
 var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };
 
+var _propTypes = require('prop-types');
+
+var _propTypes2 = _interopRequireDefault(_propTypes);
+
 var _react = require('react');
 
 var _react2 = _interopRequireDefault(_react);
@@ -18,7 +22,13 @@ var _formioUtils = require('formio-utils');
 
 var _formioUtils2 = _interopRequireDefault(_formioUtils);
 
-var _reactabular = require('reactabular');
+var _reactabularTable = require('reactabular-table');
+
+var Table = _interopRequireWildcard(_reactabularTable);
+
+var _tableResolver = require('table-resolver');
+
+var resolve = _interopRequireWildcard(_tableResolver);
 
 var _Paginator = require('./Paginator');
 
@@ -27,6 +37,8 @@ var _Paginator2 = _interopRequireDefault(_Paginator);
 var _util = require('../util');
 
 var _factories = require('../factories');
+
+function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }
 
 function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
 
@@ -198,7 +210,7 @@ var FormioGrid = function (_React$Component) {
     };
 
     _this.render = function () {
-      var rows = _reactabular.resolve.resolve({
+      var rows = resolve.resolve({
         columns: _this.state.columns,
         method: _util.nested
       })(_this.state.submissions);
@@ -206,13 +218,13 @@ var FormioGrid = function (_React$Component) {
         'div',
         { className: 'table-responsive' },
         _react2.default.createElement(
-          _reactabular.Table.Provider,
+          Table.Provider,
           {
             className: 'table table-striped table-bordered table-hover',
             columns: _this.state.columns
           },
-          _react2.default.createElement(_reactabular.Table.Header, null),
-          _react2.default.createElement(_reactabular.Table.Body, {
+          _react2.default.createElement(Table.Header, null),
+          _react2.default.createElement(Table.Body, {
             rows: rows,
             rowKey: '_id',
             onRow: _this.onRowClick
@@ -256,6 +268,6 @@ FormioGrid.defaultProps = {
   buttonLocation: 'right'
 };
 FormioGrid.propTypes = {
-  src: _react2.default.PropTypes.string
+  src: _propTypes2.default.string
 };
 exports.default = FormioGrid;

--- a/lib/components/Paginator.js
+++ b/lib/components/Paginator.js
@@ -8,6 +8,10 @@ var _extends = Object.assign || function (target) { for (var i = 1; i < argument
 
 var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();
 
+var _propTypes = require('prop-types');
+
+var _propTypes2 = _interopRequireDefault(_propTypes);
+
 var _react = require('react');
 
 var _react2 = _interopRequireDefault(_react);
@@ -144,6 +148,6 @@ var _class = function (_React$Component) {
 }(_react2.default.Component);
 
 _class.propTypes = {
-  onSelect: _react2.default.PropTypes.func
+  onSelect: _propTypes2.default.func
 };
 exports.default = _class;

--- a/package.json
+++ b/package.json
@@ -41,13 +41,14 @@
     "react-dropzone": "^3.5.3",
     "react-flatpickr": "^3.1.1",
     "react-pagify": "^2.1.1",
-    "react-quill": "^0.4.1",
+    "react-quill": "^1.2.6",
     "react-signature-canvas": "^0.3.0",
-    "react-text-mask": "^4.1.0",
+    "react-text-mask": "^5.4.3",
     "react-widgets": "^4.0.3",
-    "reactabular": "^3.0.5",
+    "reactabular-table": "^8.14.0",
     "reselect": "^2.5.4",
     "segmentize": "^0.4.1",
+    "table-resolver": "^4.1.1",
     "text-mask-addons": "^3.8.0"
   },
   "devDependencies": {
@@ -87,7 +88,7 @@
     "formio.css"
   ],
   "peerDependencies": {
-    "react": "^0.14.0 || ^15.0.0",
-    "react-dom": "^0.14.0 || ^15.0.0"
+    "react": "^0.14.9 || ^15.3.0 || ^16.0.0",
+    "react-dom": "^0.14.9 || ^15.3.0 || ^16.0.0"
   }
 }

--- a/package.json
+++ b/package.json
@@ -37,6 +37,7 @@
     "formiojs": "^2.32.3",
     "lodash": "^4.15.0",
     "moment": "^2.22.2",
+    "prop-types": "^15.7.2",
     "react-dropzone": "^3.5.3",
     "react-flatpickr": "^3.1.1",
     "react-pagify": "^2.1.1",

--- a/src/components/FormioGrid.jsx
+++ b/src/components/FormioGrid.jsx
@@ -2,10 +2,8 @@ import PropTypes from 'prop-types';
 import React from 'react';
 import Formiojs from 'formiojs';
 import FormioUtils from 'formio-utils';
-import {
-  Table, search, edit, sort, highlight, resolve,
-  SearchColumns, resizableColumn
-} from 'reactabular';
+import * as Table from 'reactabular-table';
+import * as resolve from 'table-resolver';
 import Paginator from './Paginator';
 import { nested } from '../util';
 import { FormioComponents } from '../factories';

--- a/src/components/FormioGrid.jsx
+++ b/src/components/FormioGrid.jsx
@@ -1,3 +1,4 @@
+import PropTypes from 'prop-types';
 import React from 'react';
 import Formiojs from 'formiojs';
 import FormioUtils from 'formio-utils';
@@ -27,7 +28,7 @@ export default class FormioGrid extends React.Component {
   }
 
   static propTypes = {
-    src: React.PropTypes.string
+    src: PropTypes.string
   }
 
   constructor(props) {

--- a/src/components/Paginator.jsx
+++ b/src/components/Paginator.jsx
@@ -13,7 +13,7 @@ export default class extends React.Component {
     super(props);
     this.state = {
       pagination: this.props.pagination
-    }
+    };
   }
 
   componentWillReceiveProps(nextProps) {

--- a/src/components/Paginator.jsx
+++ b/src/components/Paginator.jsx
@@ -1,3 +1,4 @@
+import PropTypes from 'prop-types';
 import React from 'react';
 import Pagify from 'react-pagify';
 import segmentize from 'segmentize';
@@ -5,7 +6,7 @@ import clone from 'lodash/clone';
 
 export default class extends React.Component {
   static propTypes = {
-    onSelect: React.PropTypes.func
+    onSelect: PropTypes.func
   }
 
   constructor(props) {

